ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/submission/CommandSubmitter.scala

```diff
@@ -175,11 +175,25 @@ case class CommandSubmitter(
       commands = commands,
       workflowId = names.workflowId,
     )
+    val commandsFirstTx = makeCommands(commands.flatMap(_.firstTransaction))
+    val commandsSecondTx = makeCommands(commands.flatMap(_.secondTransaction))
 
     (if (useSubmitAndWait) {
-       benchtoolUserServices.commandService.submitAndWait(makeCommands(commands))
+       for {
+         r <- benchtoolUserServices.commandService.submitAndWait(commandsFirstTx)
+       } yield {
+         if (commandsSecondTx.commands.nonEmpty) {
+           benchtoolUserServices.commandService.submitAndWait(commandsSecondTx)
+         } else {
+           r
+         }
+       }
+
      } else {
-       benchtoolUserServices.commandSubmissionService.submit(makeCommands(commands))
+       if (commandsSecondTx.commands.nonEmpty) {
+         sys.error("Attempting to submit two dependent transactions without using submitAndWait")
```

Comment:
I'm guessing that if tried that there could be a race condition. Preventing that by failing early